meerk40t/gui/wxmmain.py

```diff
@@ -1921,10 +1921,14 @@ def load(self, pathname):
                     channel=self.context.channel("load"),
                     svg_ppi=self.context.elements.svg_ppi,
                 )
+            # Note: I cannot see how loading a file can run commands that raise CommandSyntaxError
+            # and this exception is not raised outside commands and
+            # is not raised in svg or dxf file loading code,
+            # however this code is left in just in case.
```

Comment:
I believe this is a different bit of code intended to raise SyntaxError which should not be CommandSyntaxError but is actually using SyntaxError as a standard candle that implies malformed file. Since syntax is the thing being malformed. I'm unsure whether this is older code or the malformed parsing error is a subtype of syntax error. This is not called with the CommandSyntaxError and should not be a part of that.

Reply:
In order to correctly test this create a svg file and remove the closing brackets to cause the file to be malformed.

Reply:
I cannot see how a python syntax error could indicate a malformed data file either.

I will test what happens with a malformed file and see if there is something I need to do.

Reply:
I think ParseError might be a type of syntax error. In any event that is *not* something for the error routine within commands.

Reply:
Ok - I did as you suggested and made a malformed SVG file and got an xml.tree exception and I trapped that exception and got a decent error message.

All I need to do now is to generalise this to SVG, DXF etc. file loads (e.g. by excepting this error in svgio and raising a meerk40t bespoke BadFileError exception, and doing the same in other formats). I also need to clear the "File loading..." box before I display the error message.